packages/website/src/features/Deploy/TransactionDetailsPage.tsx

```diff
@@ -112,15 +111,17 @@ function TransactionDetailsPage() {
   // get the txn we want, we can just pluck it out of staged transactions if its there
   let safeTxn: SafeTransaction | null = null;
 
-  if (nonce < safeNonce) {
+  if (safeNonce && nonce < safeNonce) {
     // TODO: the gnosis safe transaction history is quite long, but if its not on the first page, we have to call "next" to get more txns until
     // we find the nonce we want. no way to just get the txn we want unfortunately
     // also todo: code dup
     safeTxn =
       history.results.find(
         (txn: any) =>
-          txn._nonce.toString() === nonce &&
-          (!sigHash || sigHash === getSafeTransactionHash(safe, txn))
+          txn._nonce.toString() === nonce.toString() &&
+          (!sigHash ||
+            sigHash === txn.safeTxHash ||
+            sigHash === getSafeTransactionHash(safe, txn))
```

Comment:
This line is the actual fix of the PR, all the other changes are just nesting refactors.